plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -212,19 +237,19 @@ sealed class CodeScanSessionConfig(
         private val LOG = getLogger<CodeScanSessionConfig>()
         private const val TELEMETRY_TIMEOUT_IN_SECONDS: Long = 10
         const val FILE_SEPARATOR = '/'
-        fun create(file: VirtualFile, project: Project): CodeScanSessionConfig =
+        fun create(file: VirtualFile, project: Project, scanType: String): CodeScanSessionConfig =
             when (file.programmingLanguage().toTelemetryType()) {
-                CodewhispererLanguage.Java -> JavaCodeScanSessionConfig(file, project)
-                CodewhispererLanguage.Python -> PythonCodeScanSessionConfig(file, project)
-                CodewhispererLanguage.Javascript -> JavaScriptCodeScanSessionConfig(file, project, CodewhispererLanguage.Javascript)
-                CodewhispererLanguage.Typescript -> JavaScriptCodeScanSessionConfig(file, project, CodewhispererLanguage.Typescript)
-                CodewhispererLanguage.Csharp -> CsharpCodeScanSessionConfig(file, project)
-                CodewhispererLanguage.Yaml -> CloudFormationYamlCodeScanSessionConfig(file, project)
-                CodewhispererLanguage.Json -> CloudFormationJsonCodeScanSessionConfig(file, project)
+                CodewhispererLanguage.Java -> JavaCodeScanSessionConfig(file, project, scanType)
```

Comment:
What happens if you open an unsupported language file? Does it show an error message?

Reply:
Currently yes

Reply:
That could get noisy, let's follow up to not run the scan on unsupported language files

Reply:
should i add that in follow up PR?

Reply:
Yeah follow up PR is fine

Reply:
Added in the new commit in runCoroutineScope for quick cancellation for file scans if the language is unsupported.